lib/weather/weather_report.ex

```diff
@@ -32,12 +31,9 @@ defmodule Weather.WeatherReport do
   end
 
   def handle_response({:ok, %HTTPoison.Response{ status_code: 404}}), do: { :error, "Not Found" }
-  def handle_response({:error, %HTTPoison.Error{reason: reason}}) do
-    IO.puts "#{reason}"
-  end
-  
+
   def parse_xml({xml, _}) do
-    Enum.into(@needed_fields, %{}, fn node_name -> extract_node(xml, node_name) end)
+    Enum.into(@needed_fields, %{}, &(extract_node(xml, &1)) )
```

Comment:
I think that right now, the longer form is more discoverable.  However, since the short form is more idiomatic, that's what I'm trying to go with.

My real question around this line regarded the tuple I had wrapped things in.  I was confused about how to do this, since the other instances of `handle_response` returned tuples, with a status code as the first member.

I had an epiphany last night, wherein I remembered that I could have different method signatures for `print_response`.  I deleted some code, the remaining code was simpler, and it was much more readable.